### PR TITLE
Xception Implementation

### PR DESCRIPTION
**Pull request by shubham1206agra**
Implemented Xception using the new ann-vtable structure as a Layer for more straightforward serialization and use.

### CODE REVIEW

File: models/xception/xception_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    const size_t outMaps,
+    const size_t kernelSize,
+    const size_t stride = 1,
+    const size_t padding = 0,
+    const bool useBias = false)
+{
+  block->template Add<ann::GroupedConvolution>(inMaps, kernelSize, kernelSize,
+    inMaps, stride, stride, padding, padding, "none", useBias);
+  block->template Add<ann::Convolution>(outMaps, 1, 1, 1, 1, 0, 0, "none",
+    useBias);
```
Comment (rcurtin) [inline models/xception/xception_impl.hpp:103]:
```suggestion
      inMaps, stride, stride, padding, padding, "none", useBias);
  block->template Add<ann::Convolution>(outMaps, 1, 1, 1, 1, 0, 0, "none",
      useBias);
```
Tiny style fixes, wrapped lines should be indented twice. :+1:

File: models/xception/xception_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  if (strides != 1)
+  {
+    block->template Add<ann::Padding>(1, 1, 1, 1);
+    block->template Add<ann::MaxPooling>(3, 3, strides, strides);
+  }
+  if (inMaps != outMaps || strides != 1)
+  {
+    ann::MultiLayer<MatType>* block2 = new ann::MultiLayer<MatType>();
+    block2->template Add<ann::Convolution>(outMaps, 1, 1, strides, strides,
+      0, 0, "none", false);
```
Comment (rcurtin) [inline models/xception/xception_impl.hpp:163]:
```suggestion
        0, 0, "none", false);
```
I might have missed some of these, but here's another one I saw.

**Review by rcurtin — COMMENTED** [review]
The network implementation looks good to me.  I have some pretty simple stylistic comments to handle though, but otherwise I think everything is good here. :+1:

**Review by rcurtin — APPROVED** [review]
Awesome, looks good!

**Comment by shubham1206agra** [comment]
@rcurtin Can you merge this please?

**Comment by rcurtin** [comment]
Awesome work @shubham1206agra! :+1:

**Pull request merged**